components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -1124,4 +1125,42 @@ public static boolean isUseVerifyClaimEnabled() {
         return Boolean.parseBoolean(IdentityUtil.getProperty
                 (IdentityRecoveryConstants.ConnectorConfig.USE_VERIFY_CLAIM_ON_UPDATE));
     }
+
+    /**
+     * Trigger recovery event.
+     *
+     * @param map       Property map.
+     * @param eventName Event Name.
+     * @param OTP       OTP.
+     * @throws IdentityEventException If error occurred when handleEvent.
+     */
+    public static void publishRecoveryEvent(Map<String, Object> map, String eventName, String OTP)
+            throws IdentityEventException {
+
+        Map<String, Object> eventProperties = cloneMap(map);
+        if (StringUtils.isNotEmpty(OTP)) {
+            eventProperties.put(IdentityRecoveryConstants.CONFIRMATION_CODE, OTP);
```

Comment:
If `eventProperties` map is null, it will cause a NPE. Let's add a `MapUtils.isNotEmpty` to avoid this.

Reply:
It is already handled from cloneMap method.

Reply:
cloneMap method can return a null value right?